src/InstrumentationEngine.Lib/FileLoggerSink.cpp

```diff
@@ -230,5 +230,20 @@ HRESULT CFileLoggerSink::GetPathAndFlags(_Out_ tstring* ptsPath, _Out_ LoggingFl
         *ptsPath = wszFileLogPath;
     }
 
+    return S_OK;
+}
+
+HRESULT CFileLoggerSink::SetLogFilePath(_In_ LPCWSTR wszLogFilePath)
+{
+    m_pOutputFile = nullptr; // resets the pointer & releases the FILE object.
+    m_tsPathCandidate = wszLogFilePath;
+
+    return S_OK;
+}
+
+HRESULT CFileLoggerSink::SetLogFileLevel(_In_ LoggingFlags fileLogFlags)
+{
+    m_flags = fileLogFlags;
+ 
```

Comment:
I think you need to clear m_pOutputFile and m_tsPathActual here too, otherwise the Reset method won't consider the updated flags, just in case the flag calculation happens to result in None.